docs: added first version of CONTRIBUTING.md (#52)
androiddevelopersitalia/aditogether: CONTRIBUTING.md

## How to contribute

The purpose of this page is to provide you with the minimum valuable information about how to contribute to the
project.

> ### How reach us 🗞

Reviewer: ```suggestion
### How to reach us 🗞
```

- you can join the slack community [here](https://androiddevs.it/)
- and follow the open [discussion on the repository](https://github.com/androiddevelopersitalia/aditogether/discussions)

### Set up the project ⚙️

- Have a JDK >= 11 installed (even the bundled one with AS/IJ)
- Clone the repository

> - Run `./scripts/tuner.sh` to set up the local tools used by this repository

Reviewer 2: I would add two more words here, explaining what's the tool for. An idea: 

```suggestion
- Run `./scripts/tuner.sh` to set up the local tools used by this repository: this will help to follow the guidelines and code-style, by running some checks before the commit.
```

Reviewer: @4face-studi0 the addition you proposed makes me think that the tuner scripts are running checks before the commit which is not entirely true because:
1. it doesn't run anything before the commit, it installs other scripts. This can maybe misunderstood thinking that someone has to run `tuner.sh` before committing to the project
2. it doesn't install only commit hooks

My proposal:

> - Run `./scripts/tuner.sh` to set up the local tools used by this repository: this script will perform some environmental checks and help to follow our guidelines (e.g. installing Git hooks)

> 

Reviewer 2: I would add this bit as well.
Not sure about the wording; as now, I'm referring to Detekt plugin for code-style and Kotest plugin which enables to run single (or set/suite of) Kotest test from IDE and have a nicely formatted and navigable output 

```suggestion
- Make sure to install the required IDEA plugins, from the pop-up shown when the project is opened, to have a smoother experience in the IDE

```

### Branch & commits convention 📙

We have decided to use the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), this means that there
is a specific naming convention for branch and about how to write the commits message.

For instance if you are working on a new feature the branch name must have the `feat/` prefix and then a reference about
the feature:

```
feat/1-adding-profile-section
```

The same thing for the commit message:

```shell

> git commit -m "feat(profile): implemented profile section"

Reviewer 2: See my global comment, I think it also fits here.

Reviewer: +1 on imperative form for commits

```

__Note:__ code reviews are ___mandatory___, if you are pushing your contribution, you must create a pull request.

### Lint & Static code analysis 🎨

> We love to keep our code clean and consistent, so we have decided to integrate a plugin which validate it. The detekt

Reviewer 2: Grammar-nazi alert

```suggestion
We love to keep our code clean and consistent, so we have decided to integrate a plugin which validates it. The detekt
```

configuration is available [here](https://github.com/androiddevelopersitalia/aditogether/blob/main/detekt/config.yml).

> __Note:__ if you have installed the `tuner.sh` script, all these checks are execute on pre-commit.

Reviewer 2: I've added a comment above about this, but also here it makes sense. I leave the comment so we can chose between the two options. No preference on my side

Reviewer: I think mentioning this above is fine as well to avoid repetitions.

If you prefer to mention it here as well, I'd just suggest to use `run` instead of `installed` since the tuner script can't be installed.

```suggestion
__Note:__ if you have run `./scripts/tuner.sh`, all these checks are executed on pre-commit.
```

Reviewer 2: I'm fine having it above **or** here; I would avoid duplicating it
